### DISCUSSION TITLE
Order by optional relation

### DISCUSSION THREAD

**Discussion by ArtsiomChyrkun** [prisma/prisma#19918] [Q&A]
Hi, I have a model with an optional relation. Is it possible to sort by objects, which have null?
```prisma
model School {
  code        String         @id @map("school_code")
  name        String         @map("school_name")
  address     Json
  fullAddress String         @map("full_address")
  sector      String?        @map("school_sector")
  type        String?        @map("school_type")
  contact     SchoolContact?
  geo         SchoolGeo?

  @@map("school_school_list")
}

model SchoolGeo {
  schoolCode String @id @map("school_code")
  geometry   Json

  school School? @relation(fields: [schoolCode], references: [code])

  @@map("school_school_geo")
}
```
I am getting a list of schools like this:
```ts
    const dbSchools: FullSchool[] = await prisma.school.findMany({
      include: {
        contact: true,
        geo: true,
      },
      where: {
        code: {
          in: ids,
        },
      },
      take: size,
      skip: (page - 1) * size,
    });
```
I would like smth like this:
```ts
      orderBy: {
        geo: {
          nulls: 'last'
        }
      }
```
Thanks in advance.

**Comment by nurul3101** [comment]
Hello @ArtsiomChyrkun 👋 

You are looking for [OrderByNulls](https://www.prisma.io/docs/concepts/components/prisma-client/filtering-and-sorting#sort-with-null-records-first-or-last) feature. 
This feature is generally available in version 4.16.0 and later. To use this feature in versions [4.1.0](https://github.com/prisma/prisma/releases/tag/4.1.0) to [4.15.0](https://github.com/prisma/prisma/releases/tag/4.15.0) the [Preview feature](https://www.prisma.io/docs/concepts/components/preview-features/client-preview-features#enabling-a-prisma-client-preview-feature) `orderByNulls` will need to be enabled.

In your case, the query would look something like this:
```ts
  await prisma.school.findMany({
    include: {
      geo: true,
    },
    where: {
      code: {
        in: ids,
      },
    },
    take: size,
    skip: (page - 1) * size,
    orderby: {
      geo: {
        sort: 'desc',
        nulls: 'last',
      },
    },
  });
```

**Reply by ArtsiomChyrkun** [reply]
Hi. It won't work, becase geo is an included type.
The generated type for ordering looks like this:
```
  export type SchoolGeoOrderByWithRelationInput = {
    schoolCode?: SortOrder
    geometry?: SortOrder
    school?: SchoolOrderByWithRelationInput
  }

  export const SortOrder: {
    asc: 'asc',
    desc: 'desc'
  };

  export type SortOrder = (typeof SortOrder)[keyof typeof SortOrder]
```

I  guess there is no solution to sort by empty relations
